homeassistant/components/nobo_hub/climate.py

```diff
@@ -95,8 +91,37 @@ def __init__(self, zone_id, hub: nobo, override_type) -> None:
             via_device=(DOMAIN, hub.hub_info[ATTR_SERIAL]),
             suggested_area=hub.zones[zone_id][ATTR_NAME],
         )
+        self._supports_temp_modification = self.supports_temp_modification
+        self._attr_supported_features = self.supported_features
         self._read_state()
 
+    @property
+    def supports_temp_modification(self) -> dict[str, bool]:
+        """Determine which presets support temperature modification by checking zone components."""
+        supports_modification = {
+            nobo.API.NAME_COMFORT: False,
+            nobo.API.NAME_ECO: False,
+        }
+        for component in self._nobo.components.values():
+            if component["zone_id"] == self._id:
+                supports_modification[nobo.API.NAME_COMFORT] |= component[
+                    "model"
+                ].supports_comfort
+                supports_modification[nobo.API.NAME_ECO] |= component[
+                    "model"
+                ].supports_eco
+        return supports_modification
+
+    @property
+    def supported_features(self) -> ClimateEntityFeature:
+        """Determine supported climate entity features."""
+        supported_features = ClimateEntityFeature.PRESET_MODE
+        if all(self._supports_temp_modification):
+            supported_features |= ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+        elif any(self._supports_temp_modification):
+            supported_features |= ClimateEntityFeature.TARGET_TEMPERATURE
```

Comment:
I think the calls to `all` + `any` here are to fix a previous review comment suggesting to "simplify" the code.
However, I think the current iteration is difficult to understand and I wonder if it really works as intended?
To my understanding, calling `all` on a dict will return `True` if all keys are truthy, which they will always be since they're non-empty strings.


I'd suggest to revert this commit https://github.com/home-assistant/core/pull/145157/commits/28526eac7782bed68d2f17b2f10359458aeaa8d0, it's not an improvement.